### PR TITLE
fix: siwe map to siwx onSignOut callback

### PR DESCRIPTION
**Pull request by zoruka**
# Description

- Fix `onSignOut` not being called for `activeCaipAddress` changes;
- Fix `signOut`/`onSignOut` not being called on `useDisconnect` hook calls;
- Fix `AppKitSIWEClient` `signIn` and `signOut` methods to call new SIWX handlers;
- Add tests for `mapToSIWX` function and usage against AppKit.

## Type of change

- [ ] Chore (non-breaking change that addresses non-functional tasks, maintenance, or code quality improvements)
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

# Checklist

- [x] Code in this PR is covered by automated tests (Unit tests, E2E tests)
- [x] My changes generate no new warnings
- [x] I have reviewed my own code
- [x] I have filled out all required sections
- [x] I have tested my changes on the preview link
- [ ] Approver of this PR confirms that the changes are tested on the preview link

### CODE REVIEW

File: packages/siwe/src/mapToSIWX.ts [hunk 1]
```diff
@@ -19,37 +21,51 @@ export function mapToSIWX(siwe: AppKitSIWEClient): SIWXConfig {
...
+      const session = await getSession()
+      const isDifferentNetwork =
+        session &&
+        session.chainId !== NetworkUtil.caipNetworkIdToNumber(activeCaipNetwork?.caipNetworkId)
 
-  ChainController.subscribeKey('activeCaipAddress', async activeCaipAddress => {
-    if (!siwe.options.signOutOnAccountChange) {
-      return
-    }
+      if (isDifferentNetwork) {
```
Comment (Cali93) [inline packages/siwe/src/mapToSIWX.ts:41]: I think you should only sign out the user if `signOutOnNetworkChange` is true

File: packages/siwe/src/mapToSIWX.ts [hunk 2]
```diff
@@ -19,37 +21,51 @@ export function mapToSIWX(siwe: AppKitSIWEClient): SIWXConfig {
...
+      }
 
-    const compareSessionAddress = session?.address.toLowerCase()
-    const compareCaipAddress = CoreHelperUtil?.getPlainAddress(activeCaipAddress)?.toLowerCase()
+      if (siwe.options.signOutOnAccountChange) {
+        const session = await getSession()
 
-    const isDifferentAddress = session && compareSessionAddress !== compareCaipAddress
+        const compareSessionAddress = session?.address.toLowerCase()
+        const compareCaipAddress = CoreHelperUtil?.getPlainAddress(activeCaipAddress)?.toLowerCase()
```
Comment (Cali93) [inline packages/siwe/src/mapToSIWX.ts:59]: nit: variable names seem off

File: packages/siwe/src/mapToSIWX.ts [hunk 3]
```diff
@@ -19,37 +21,51 @@ export function mapToSIWX(siwe: AppKitSIWEClient): SIWXConfig {
...
+          await signOut()
+        }
 
-    const session = await getSession()
+        return
+      }
 
-    const compareSessionAddress = session?.address.toLowerCase()
-    const compareCaipAddress = CoreHelperUtil?.getPlainAddress(activeCaipAddress)?.toLowerCase()
+      if (siwe.options.signOutOnAccountChange) {
```
Comment (Cali93) [inline packages/siwe/src/mapToSIWX.ts:55]: What if the param isn't provided. Based on docs it should default to `true`.
https://docs.reown.com/appkit/next/core/siwe#signoutondisconnect-boolean---defaults-to-true

Comment (tomiir) [reply packages/siwe/src/mapToSIWX.ts:55]: shouldn't this be set during initialization as a default value?

File: packages/siwe/src/mapToSIWX.ts (hunk 1 above)
Comment (zoruka) [reply packages/siwe/src/mapToSIWX.ts:41]: There is a condition right above checking if `signOutOnNetworkChange` is false then it stops the function

**Review by Cali93 — APPROVED** [review]
Nice 👌

**Pull request merged**